javascript/0912-sort-an-array.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * https://leetcode.com/problems/sort-an-array/
+ * Merge Sort
+ * Time O(n*log(n)) | Space O(n)
+ * @param {number[]} nums
+ * @return {number[]}
+ */
+var sortArray = function(nums) {
+    
+    const merge = (left, mid, right) => {
```

Comment:
Suggestion: Avoid using global scope with nested functions

Reply:
I don't understand. What do you mean. As far as I can understand You don't want me to use nums variable  inside the merge (nested function). But still I need to confirm this. The reason I have nested the merge function inside sortArray is because I don't want to pass the  nums veriable again and again.

Reply:
It’s ok to add as a new parameter to avoid using global scope.

```js
const sortArray = (nums, left = 0, right = (nums.length - 1)) =>
    mergeSort(nums, left, right);

const merge = (nums, left, right) => { … }

const sort = (nums, left, right, mid) => { … }

```

Reply:
Hmm, I have created each function outside of the main function. Let me know if You would like to make further changes.